tests/unit/framework/test_crawlers.py

```diff
@@ -51,6 +51,28 @@ def test_full_name():
     assert "a.b.c" == cb._full_name
 
 
+def test_escaped_path():
+    cb = CrawlerBase(MockBackend(), "a", "b", "c", Bar)
+    assert "a.b.c" == cb.escape(cb._full_name)
+    # test with optional escaping
+    assert "a" == cb.escape("a")
+    assert "a.b" == cb.escape("a.b")
+    assert "a.b.c" == cb.escape("a.b.c")
+    assert "a" == cb.escape("a")
+    assert "a.b" == cb.escape("a.b")
+    assert "`a`.`b`.`c`" == cb.escape("`a`.`b`.`c`")
+    assert "`a-b`.c.d" == cb.escape("a-b.c.d")
+    assert "a.`b-c`.d" == cb.escape("a.b-c.d")
+    assert "a.b.`c-d`" == cb.escape("a.b.c-d")
+    # test with escaping enforced
+    assert "`a`" == cb.escape("a", False)
+    assert "`a`.`b`" == cb.escape("a.b", False)
+    assert "`a`.`b`.`c`" == cb.escape("a.b.c", False)
+    assert "`a-b`.`c`.`d`" == cb.escape("a-b.c.d", False)
+    assert "`a`.`b-c`.`d`" == cb.escape("a.b-c.d", False)
+    assert "`a`.`b`.`c-d`" == cb.escape("a.b.c-d", False)
+
```

Comment:
Amazing!!!
Please consider testing unicode, kanji, embedded space characters.

This documentation isn't much help... https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html#names